### PR TITLE
Flutter 3 build fix

### LINKED ISSUE #105: Beta2  still doesn't compile on Flutter 3
hi! looks like there's stll an error when using Flutter 3

```
e: /Users/rafal.wachol/.pub-cache/hosted/pub.dartlang.org/datadog_flutter-2.0.0-beta.2/android/src/main/kotlin/com/greenbits/datadog_flutter/DatadogFlutterPlugin.kt: (35, 5): Unresolved reference: context
e: /Users/rafal.wachol/.pub-cache/hosted/pub.dartlang.org/datadog_flutter-2.0.0-beta.2/android/src/main/kotlin/com/greenbits/datadog_flutter/DatadogFlutterPlugin.kt: (63, 20): Unresolved reference: context

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':datadog_flutter:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

```

Looks like companion object was removed together with `context` variable:

https://github.com/GetDutchie/datadog_flutter/blob/8f38c649d65e816b57dbf4c56dbd44b0e2d70513/android/src/main/kotlin/com/greenbits/datadog_flutter/DatadogFlutterPlugin.kt#L50

### PR DESCRIPTION
**Pull request by charafau**
Add removed context variable
Applied kotlin lint suggestions
Removed deprecated API

fixes: https://github.com/GetDutchie/datadog_flutter/issues/105

### CODE REVIEW

File: android/src/main/kotlin/com/greenbits/datadog_flutter/DatadogFlutterPlugin.kt [hunk 1]
```diff
@@ -15,286 +14,287 @@ import com.datadog.android.core.configuration.Credentials
...
-      call.method == "initWithClientToken" -> {
-        val rumApplicationId = call.argument<String>("androidRumApplicationId")
-        var configBuilder = Configuration.Builder(
-          true,
-          true,
-          rumApplicationId != null,
-          rumApplicationId != null
-        )
-        if (call.argument<Boolean>("useEUEndpoints")!!) {
-          configBuilder = configBuilder.useEUEndpoints()
```
Comment (charafau) [inline android/src/main/kotlin/com/greenbits/datadog_flutter/DatadogFlutterPlugin.kt:51]: This was deprecated

File: android/src/main/kotlin/com/greenbits/datadog_flutter/DatadogFlutterPlugin.kt [hunk 2]
```diff
@@ -15,286 +14,287 @@ import com.datadog.android.core.configuration.Credentials
...
-public class DatadogFlutterPlugin: FlutterPlugin, MethodCallHandler {
+class DatadogFlutterPlugin: FlutterPlugin, MethodCallHandler {
   private var loggers = mutableMapOf<String, Logger>()
   private var traces = mutableMapOf<String, Span>()
+  private lateinit var context : Context
+  private lateinit var channel : MethodChannel
 
   override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
-    context = flutterPluginBinding.getApplicationContext()
-    val channel = MethodChannel(flutterPluginBinding.getFlutterEngine().getDartExecutor(), "plugins.greenbits.com/datadog_flutter")
```
Comment (charafau) [inline android/src/main/kotlin/com/greenbits/datadog_flutter/DatadogFlutterPlugin.kt:36]: This was also deprecated so I moved to new way

**Comment by tshedor** [comment]
Hey @charafau this is really good work and I appreciate the cleanup. Is there anyway you can only PR the changes required to build for Flutter 3?

**Comment by charafau** [comment]
@tshedor  sorry, went ahead of myself :) there you go https://github.com/GetDutchie/datadog_flutter/pull/107

**Comment by charafau** [comment]
@tshedor  Since android fix is in now, should I push code cleanup changes?

**Pull request closed without merging**